src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -371,11 +371,16 @@ public static TDelegate TryCompile<TDelegate>(this ExpressionInfo<TDelegate> lam
                 typeof(ExpressionCompiler), skipVisibility: true);
 
             var il = method.GetILGenerator();
-            if (!EmittingVisitor.TryEmit(exprObj, exprNodeType, exprType, paramExprs, il, 
+            if (!EmittingVisitor.TryEmit(exprObj, exprNodeType, exprType, paramExprs, il,
                 ref closureInfo, ExpressionType.Default))
                 return null;
 
-            if (returnType == typeof(void) && exprType != typeof(void))
+            if (returnType == typeof(void) && exprType != typeof(void)
+                &&
+                // TODO: fix it, why body has return of ref if there is, if 2 refs then gots object
+                (paramTypes.GetFirstIndex(x => x.IsByRef) <= 0

```

Comment:
Let's move these two by-ref checks to separate method, and consolidate and inline the `GetFirst..` checks. Inline them into normal for loop with check and return, without lambdas.

Reply:
I did not know that git shows changes of my push even without new pull request. Will do inline if no other way found, right now I am looking into parameters of method. May be there is some other hint to resolve logic. Current behavior of C# weird - there is return if there is ref.
```
return void + ref = ret type of ref
return T + ref = ret type is of T
void + ref + ref = object
```
So return we see in C# is not return which expression think.